coderd/workspaces.go

```diff
@@ -294,28 +294,39 @@ func (api *API) postWorkspacesByOrganization(rw http.ResponseWriter, r *http.Req
 		return
 	}
 
-	workspace, err := api.Database.GetWorkspaceByOwnerIDAndName(ctx, database.GetWorkspaceByOwnerIDAndNameParams{
+	workspaces, err := api.Database.GetWorkspaces(ctx, database.GetWorkspacesParams{
 		OwnerID: user.ID,
-		Name:    createWorkspace.Name,
 	})
-	if err == nil {
-		// If the workspace already exists, don't allow creation.
-		httpapi.Write(ctx, rw, http.StatusConflict, codersdk.Response{
-			Message: fmt.Sprintf("Workspace %q already exists.", createWorkspace.Name),
-			Validations: []codersdk.ValidationError{{
-				Field:  "name",
-				Detail: "This value is already in use and should be unique.",
-			}},
-		})
-		return
-	}
 	if err != nil && !errors.Is(err, sql.ErrNoRows) {
 		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
 			Message: fmt.Sprintf("Internal error fetching workspace by name %q.", createWorkspace.Name),
 			Detail:  err.Error(),
 		})
 		return
 	}
+	for _, workspace := range workspaces {
+		if workspace.Name == createWorkspace.Name {
+			// If the workspace already exists, don't allow creation.
+			httpapi.Write(ctx, rw, http.StatusConflict, codersdk.Response{
+				Message: fmt.Sprintf("Workspace %q already exists.", createWorkspace.Name),
+				Validations: []codersdk.ValidationError{{
+					Field:  "name",
+					Detail: "This value is already in use and should be unique.",
+				}},
+			})
+			return
+		}
+	}
```

Comment:
It might be a bit cleaner to get a count instead of iterating. It's not unreasonable for a single user to have hundreds or even thousands (if the user is being automated or something).

Reply:
That's fair, I can do a count query to save the db